pinot-common/src/main/java/org/apache/pinot/common/utils/TarCompressionUtils.java

```diff
@@ -66,106 +67,122 @@ public class TarGzCompressionUtils {
       throws IOException {
     String entryName = baseEntryName + file.getName();
     TarArchiveEntry entry = new TarArchiveEntry(file, entryName);
-    tarGzOut.putArchiveEntry(entry);
+    tarOut.putArchiveEntry(entry);
     if (file.isFile()) {
       try (InputStream in = Files.newInputStream(file.toPath())) {
-        IOUtils.copy(in, tarGzOut);
+        IOUtils.copy(in, tarOut);
       }
-      tarGzOut.closeArchiveEntry();
+      tarOut.closeArchiveEntry();
     } else {
-      tarGzOut.closeArchiveEntry();
+      tarOut.closeArchiveEntry();
 
       File[] children = file.listFiles();
       assert children != null;
       String baseEntryNameForChildren = entryName + ENTRY_NAME_SEPARATOR;
       for (File child : children) {
-        addFileToTarGz(tarGzOut, child, baseEntryNameForChildren);
+        addFileToCompressedTar(tarOut, child, baseEntryNameForChildren);
       }
     }
   }
 
   /**
-   * Un-tars a tar.gz file into a directory, returns all the untarred files/directories.
+   * Un-tars a compressed tar file into a directory, returns all the untarred files/directories.
    * <p>For security reason, the untarred files must reside in the output directory.
    */
   public static List<File> untar(File inputFile, File outputDir)
-      throws IOException {
+      throws IOException, CompressorException {
     try (InputStream fileIn = Files.newInputStream(inputFile.toPath())) {
       return untar(fileIn, outputDir);
     }
   }
 
   /**
-   * Un-tars an inputstream of a tar.gz file into a directory, returns all the untarred files/directories.
+   * Un-tars an inputstream of a compressed tar file into a directory, returns all the untarred files/directories.
    * <p>For security reason, the untarred files must reside in the output directory.
    */
   public static List<File> untar(InputStream inputStream, File outputDir)
-      throws IOException {
+      throws IOException, CompressorException {
     return untarWithRateLimiter(inputStream, outputDir, NO_DISK_WRITE_RATE_LIMIT);
   }
 
   /**
-   * Un-tars an inputstream of a tar.gz file into a directory, returns all the untarred files/directories.
+   * Un-tars an inputstream of a compressed tar file into a directory, returns all the untarred files/directories.
    * RateLimit limits the untar rate
    * <p>For security reason, the untarred files must reside in the output directory.
    */
   public static List<File> untarWithRateLimiter(InputStream inputStream, File outputDir, long maxStreamRateInByte)
-      throws IOException {
+      throws IOException, CompressorException {
     String outputDirCanonicalPath = outputDir.getCanonicalPath();
     // Prevent partial path traversal
     if (!outputDirCanonicalPath.endsWith(File.separator)) {
       outputDirCanonicalPath += File.separator;
     }
     List<File> untarredFiles = new ArrayList<>();
     try (InputStream bufferedIn = new BufferedInputStream(inputStream);
```

Comment:
(format) Please follow [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#set-up-ide), same for other places

Reply:
Switch to using Pinot Style and adjusted style in various files.